pkg/util/azure/azure_tags.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package azure
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type azureInstanceMetadata struct {
+	VMID          string `json:"vmId"`
+	Zone          string `json:"zone"`
+	VMSize        string `json:"vmSize"`
+	ResourceGroup string `json:"resourceGroupName"`
+}
+
+// GetTags gets the tags from the GCE api
+func GetTags() ([]string, error) {
+	tags := []string{}
+
+	metadataResponse, err := getResponse(metadataURL + "/metadata/instance/compute?api-version=2017-08-01")
+	if err != nil {
+		return tags, fmt.Errorf("unable to query metadata endpoint: %s", err)
+	}
+
+	metadata := azureInstanceMetadata{}
+
+	err = json.Unmarshal([]byte(metadataResponse), &metadata)
+	if err != nil {
+		return tags, err
+	}
+
+	if metadata.VMID != "" {
+		tags = append(tags, fmt.Sprintf("vm-id:%s", metadata.VMID))
+	}
```

Comment:
The crawler does not appear to tag the VM id. It does use it to determine the host name though. The VM id may show up as hostname if no other name is determined.

Reply:
@graybjones I'm not familiar with the crawler you're referring to?

Reply:
Sorry for the confusion. I will edit my comments for clarity.

Reply:
I don't see a way to currently know whether or not the hostname has been set to determine whether or not to include this tag. Any suggestions?